### PR TITLE
Refactor client (remove landing view) and switch microphone is broken in room view [ECO-5813]

### PR DESCRIPTION
**Pull request by emilianop11**

### CODE REVIEW

**Review by Raulkg — COMMENTED** [review]
I am getting console.logs on the browser as below:

requests.js:36 GET http://localhost:8123/room/old-feather-shiny-block-9866/rawInfo 404 (Not Found)
(anonymous) @ requests.js:36
sendXHR @ requests.js:8
getRoomRawInfo @ requests.js:56
(anonymous) @ precallController.js:106

**Comment by emilianop11** [comment]
> I am getting console.logs on the browser as below:
> 
> requests.js:36 GET http://localhost:8123/room/old-feather-shiny-block-9866/rawInfo 404 (Not Found)
> (anonymous) @ requests.js:36
> sendXHR @ requests.js:8
> getRoomRawInfo @ requests.js:56
> (anonymous) @ precallController.js:106

That is expected. It just means that the room you are trying to join does not exist yet.

**Comment by Raulkg** [comment]
> > I am getting console.logs on the browser as below:
> > requests.js:36 GET http://localhost:8123/room/old-feather-shiny-block-9866/rawInfo 404 (Not Found)
> > (anonymous) @ requests.js:36
> > sendXHR @ requests.js:8
> > getRoomRawInfo @ requests.js:56
> > (anonymous) @ precallController.js:106
> 
> That is expected. It just means that the room you are trying to join does not exist yet.

i don't think we should throw an error in console logs. we should handle errors like this using try catch

File: web/js/precallController.js [hunk 1]
```diff
@@ -93,26 +83,28 @@
...
+        function hidePrecall(roomName) {
           PrecallView.hide();
           publisher && publisher.destroy();
           otNetworkTest && otNetworkTest.stopTest();
+          const username = document.getElementById('user-name-input').value.trim();
+          window.localStorage.setItem('username', username);
+          publisherOptions.name = username;
           setTimeout(() => {
             resolve({
+              roomURI: roomName,
```
Comment (Raulkg) [inline web/js/precallController.js:95]: roomName variable is also declared in upper scope. Consider changing the variable name to avoid confusion.

**Review by Raulkg — APPROVED** [review]
I will address the issue of critical js with a different patch PR.

for now this LGTM

**Review by jeffswartz — CHANGES_REQUESTED** [review]

**Review by jeffswartz — CHANGES_REQUESTED** [review]
There's a reference to views/index.ejs in Gruntfile.js. And in CUSTOMIZING-UI.md.

**Review by jeffswartz — APPROVED** [review]

**Pull request merged**